src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+        }
+        Test.startTest();
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update contacts;
+        Test.stopTest();
+
+        // verify results
+        contacts = [select Id, Name, MailingStreet, MailingCity, MailingState, MailingPostalCode, MailingCountry, 
+                   Current_Address__c from Contact order by Contact.Name];
+        system.assertEquals(contactCount, contacts.size());
+        
+        List<Address__c> ListAddr = [select Id, Default_Address__c, MailingStreet__c, MailingCity__c, 
+        Parent_Contact__c, Latest_Start_Date__c, Latest_End_Date__c from Address__c 
+        order by Parent_Contact__r.Name];
+        system.assertEquals(contactCount, ListAddr.size());
+                
+        for (Integer i = 0; i < contactCount; i++) {
+            Contact contact = contacts[i];
+            Address__c addr = ListAddr[i];
+            system.assertEquals(contact.MailingStreet, addr.MailingStreet__c);
+            system.assertEquals(contact.MailingCity, addr.MailingCity__c);
+            system.assertNotEquals(null, contact.MailingStreet);
+            system.assertNotEquals(null, contact.MailingCity);
+            system.assertEquals(true, addr.Default_Address__c);
+            system.assertNotEquals(null, contact.Current_Address__c);
+        }
+    }
+    
+    /*********************************************************************************************************
+    @description Update N Contacts with address objects 
+    verify:
+        Contact addresses updated
+        contact addresses not updated
+    **********************************************************************************************************/            
+    @isTest
+    public static void newAddrForContacts() {           
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c(Contacts_Addresses_Enabled__c = true));
+ 
+        // create Contacts without addresses
+        List<Contact> contacts = UTIL_UnitTestData_TEST.CreateMultipleTestContacts(3);
+        insert contacts;
+
+         // create addresses for the Contacts
+        List<Address__c> addrs = UTIL_UnitTestData_TEST.initTestAddr(contacts.size());
+        for (Integer i = 0; i < contacts.size(); i++) {
+            addrs[i].Parent_Contact__c = contacts[i].Id;
+            addrs[i].Default_Address__c = true;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        insert addrs;
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        
+        // verify results
+        contacts = [select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c 
+        from Contact order by Contact.Name];
+        system.assertEquals(3, contacts.size());
+        
+        addrs = [select Id, Default_Address__c, MailingStreet__c, MailingCity__c, Parent_Contact__c, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause or contain the number of records redurned, or adding a `LIMIT` clause to cap the maximum number of records returned.